python/ray/actor.py

```diff
@@ -438,8 +438,13 @@ def _remote(self,
                 concurrency defaults to 1 for threaded execution, and 1000 for
                 asyncio execution. Note that the execution order is not
                 guaranteed when max_concurrency > 1.
-            name: The globally unique name for the actor.
-            detached: DEPRECATED.
+            name: The globally unique name for the actor, which can be used
+                to retrieve the actor via ray.get_actor(name) as long as the
+                actor is still alive.
+            lifetime: Either "default", which specifies the actor will fate
+                share with its creator and will be deleted once its refcount
+                drops to zero, or "detached", which means the actor will live
```

Comment:
I'm a bit concerned with having named actors be refcounted normally. To me, it seems like the name acts as a global reference so the actor should exist until either the parent exits or the job finishes.

Note that in its current form this change would not be very useful for Serve but with the suggested change it would allow us to use the existing codepath to implement Serve instances that are scoped to a driver program by only changing a single flag when creating actors. With the ref counting, the code would need to be restructured to keep actor handles to all of the created actors.

Reply:
Not that we should just make the API whatever is best for Serve, but I think it's sort of the only datapoint that we have for something that wants to make use of both lifetime options.

Reply:
This is actually useful for RLlib though, which uses global named actors but does want fate sharing. The name change was a bit of a regression there actually.